include/nlohmann/detail/output/binary_writer.hpp

```diff
@@ -1813,8 +1813,18 @@ class binary_writer
                enable_if_t < std::is_signed<C>::value && std::is_unsigned<char>::value > * = nullptr >
     static CharType to_char_type(std::uint8_t x) noexcept
     {
-        static_assert(sizeof(std::uint8_t) == sizeof(CharType), "size of CharType must be equal to std::uint8_t");
+        // Workaround missing "is_trivially_copyable" &&
+        // "is_trivially_default_constructible" in multiple older compiler.
+        // Fallback to "is_trivial" check if C++26 not available.
+        // See <https://github.com/nlohmann/json/issues/4778>.
```

Comment:
Can we word this as something like the below?

The `is_trivial` function is deprecated in C++26.  The replacement is to use `is...` and `is...`.  However, some older libraries that implement `is_trivial` don't implement those functions, and it's not easy to detect all of them, so use `is_trivial` unless we are using a standard where it has been deprecated.

Reply:
I agree with this and https://github.com/nlohmann/json/pull/4775#issuecomment-2884361627.